README.md

```diff
@@ -53,14 +54,18 @@ input.addEventListener('auto-check-send', function(event) {
   console.log('Adding to FormData before network request is sent.')
   const {body} = event.detail
   body.append('custom_form_data', 'value')
+
+  // Invalidate input while network request is live.
+  event.detail.setValidity('Checking with server…')
 })
-input.addEventListener('auto-check-success', function(event) {
-  const {message} = event.detail
+input.addEventListener('auto-check-success', async function(event) {
+  const message = await event.detail.response.text()
   console.log('Validation passed', message)
 })
 input.addEventListener('auto-check-error', function(event) {
-  const {message} = event.detail
-  console.log('Validation failed', message)
+  // Asynchronously extract text response and invalidate the input.
```

Comment:
Same here.

```suggestion
  // Asynchronously extract text response and provide a custom validation message for the input.
```

Reply:
IMO it'd be clearer if we document these in like `<text-expander>` where each event is listed, including its custom detail and their types. Specifically so we can write in words that `setValidity` takes a `Promise<string>` or `string`.

https://github.com/github/text-expander-element#events